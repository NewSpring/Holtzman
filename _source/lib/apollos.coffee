###

  Apollos.name

  @example get a string with the name of this system

    console.log Apollos.name

###
Apollos.name = "Apollos"


###

  Apollos.user

  @example get the currently logged in user

    console.log Apollos.user()._id

###
Apollos.user = (id) ->

  if Meteor.isServer
    return {}

  if Meteor.isClient
    user = Meteor.user()

  return user or {}


Apollos.user.login = {}

if Meteor.isClient
  Meteor.subscribe "currentUserPerson"
else
  Meteor.publish "currentUserPerson", ->
    user = Apollos.users.findOne @.userId
    query = {}
    queryOk = false

    if not user
      return

    if user.personGuid
      query.guid = user.personGuid
      queryOk = true

    # we need to move the rock stuff out of this package
    if user.rock and user.rock.personId
      query.personId = user.rock.personId
      queryOk = true

    if not queryOk
      return

    return Apollos.people.find query


###

  Apollos.person

  @example get the currently logged in user's person document

    console.log "Hello, #{Apollos.person().firstName}"

###
Apollos.person = (user) ->

  if Meteor.isServer
    if not user?._id
      throw new Meteor.Error ("user is requried for server lookup")
      return

  if Meteor.isClient
    user or= Apollos.user()

  if user.personGuid
    person = Apollos.people.findOne
      guid: user.personGuid


  # we shouldnt have any code for rock in the core package
  # this will probably need to adjust the rock package

  # if not person and user.rock and user.rock.personId
  #   person = Apollos.people.findOne
  #     personId: user.rock.personId

  return person or {}






if Meteor.server
  Accounts.onCreateUser (options, user) ->

    if options.profile
      user.profile = options.profile

    if user.profile?.guest is true
      return user

    person = Apollos.person user

    # # no existing user so create one
    if not Object.keys(person).length
      if not user.personGuid
        user.personGuid = Apollos.utilities.makeNewGuid()

      Apollos.people.upsert({guid: user.personGuid}, {
        $set:
          preferredEmail: user.emails[0].address
      })

    return user

###

  Apollos.user.create

  @example create a new user

    Apollos.user.create "xyz@abc.cc", "password1", (error) ->
      if error
        console.log error

  @param email is the email address for the user to use to login with
  @param password is the plain-text password the user will authenticate with
  @param callback is the function that will be called with an error if so

###
Apollos.user.create = (email, password, callback) ->

  Accounts.createUser
    email: email
    password: password
  ,
    callback

###

  Apollos.user.forgotPassword

  @example request a forgot password email

    Apollos.user.forgotPassword "xyz@abc.cc", (error) ->
      if error
        console.log error

  @param email is the email address for the user to use to login with
  @param callback is the function that will be called with an error if so

###
Apollos.user.forgotPassword = (email, callback) ->

  return Accounts.forgotPassword
    email: email
  ,
    callback

###

  Apollos.user.resetPassword

  @example changes password using reset token

    Apollos.user.resetPassword token, newPassword, (error) ->
      if error
        console.log error

  @param token is the reset token emailed to the user
  @param newPassword is what to change the password to
  @param callback is the function that will be called with an error if so

###
Apollos.user.resetPassword = (token, newPassword, callback) ->

  return Accounts.resetPassword token, newPassword, callback


###

  Apollos.user.changePassword

  @example changes password using old password

    Apollos.user.changePassword oldPassword, newPassword, (error) ->
      if error
        console.log error

  @param oldPassword is the user's current password
  @param newPassword is what to change the password to
  @param callback is the function that will be called with an error if so

###
Apollos.user.changePassword = (oldPassword, newPassword, callback) ->

  return Accounts.changePassword oldPassword, newPassword, callback

if Meteor.isServer
  # TODO this is not secure
  Apollos.emailTemplates = Accounts.emailTemplates
