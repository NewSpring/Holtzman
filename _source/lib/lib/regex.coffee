_addRegexTestFunction = (regex) ->
  funcName = "is#{regex.charAt(0).toUpperCase()}#{regex.slice(1)}"
  Apollos.validate[funcName] = (str) ->
    return Apollos.regex[regex].test str


Apollos.regex =

  # @TODO choose which email regex to use
  #
  # this was ours
  # email: /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
  #
  # this is what rock uses:
  email: /[\w\.\'_%-]+(\+[\w-]*)?@([\w-]+\.)+[\w-]+/

  bcrypt: /^\$2a\$10\$[\/\.a-zA-Z0-9]{53}$/

  phoneNumber: /^[1-9]([0-9]{6}|[0-9]{9})$/

  guid: /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/i

  visa: /^4[0-9]{12}(?:[0-9]{3})?$/

  mastercard: /^5[1-5][0-9]{14}$/

  americanExpress: /^3[47][0-9]{13}$/

  discover: /^6(?:011|5[0-9]{2})[0-9]{12}$/

  startOfVisa: /^4[0-9]{0,15}$/

  startOfMastercard: /^5$|^5[1-5][0-9]{0,14}$/

  startOfAmEx: /^3$|^3[47][0-9]{0,13}$/

  startOfDiscover: /^6$|^6[05]$|^601[1]?$|^65[0-9][0-9]?$|^6(?:011|5[0-9]{2})[0-9]{0,12}$/


Apollos.addRegex = (obj) ->

  for key, value of obj
    Apollos.regex[key] = value

  return


Apollos.validate =

  acceptedCardTypes: ["Visa", "Mastercard", "AmEx", "Discover"]

  guessCardType: (number) ->

    number = number.replace /-/g, ''

    for cardType in Apollos.validate.acceptedCardTypes

      if Apollos.validate["isStartOf#{cardType}"] number
        return cardType

    return false


  getCardType: (number) ->

    number = number.replace /-/g, ''

    for cardType in Apollos.validate.acceptedCardTypes

      if Apollos.validate["is#{cardType}"] number
        return cardType

    return false


  isBcryptHash: (str) ->
    return @.isBcrypt(str)


for regex of Apollos.regex
  _addRegexTestFunction regex
