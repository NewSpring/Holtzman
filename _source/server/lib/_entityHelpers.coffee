Apollos.documentHelpers =


  translate: (singular, data, platform) ->

    if not platform
      Apollos.debug "must specify platform to translate to"
      return

    platform = platform.toUpperCase()

    if not Apollos[singular]._dictionary[platform]
      Apollos.debug "no translation found for #{platform} in #{singular}"
      return

    # translate
    return Apollos[singular]._dictionary[platform] data


  ###

    Apollos.documentHelpers.update

    @example update a doc in apollos with data from a platform

      Apollos.documentHelpers.update "person", "people", person, platform

    @param singular [String] The singular form of the entities name
    @param plural [String] The plural form of the entities name
    @param doc [Object] existing doc from other service to be updated
    @param platform [String] platform to be updated from

  ###
  update: (singular, plural, doc, platform) ->

    doc = Apollos[singular].translate doc, platform

    if not doc
      return

    singularIdKeyValue = {}
    singularIdKeyValue["#{singular}Id"] = doc["#{singular}Id"]

    if platform
      doc.updatedBy = platform.toUpperCase()
    else
      doc.updatedBy = Apollos.name

    query =
      $or: [
        singularIdKeyValue
      ,
        guid: RegExp(doc.guid, "i")
      ]

    matches = Apollos[plural].find query,
      sort:
        updatedDate: 1

    if matches.count() > 1
      # Delete older documents, which are the first ones since they are sorted
      ids = matches.map (m) ->
        return m._id

      ids.pop()

      Apollos[plural].remove
        _id:
          $in: ids

      matches = Apollos[plural].find(query)

    if matches.count() is 1
      existing = matches.fetch()[0]
      mongoId = existing._id

      Apollos[plural].update
        _id: mongoId
      ,
        $set: doc

    else
      mongoId = Apollos[plural].insert doc

    return mongoId


  ###

    Apollos.documentHelpers.delete

    @example take an colection and delete it

      Apollos.documentHelpers.delete("person", "people", 3, plaform)

    @param singular [String] The singular form of the entities name
    @param plural [String] The plural form of the entities name
    @param identifier [Object|String|Number] existing document, _id
    @param platform [String] platform initiating the delete

  ###
  delete: (singular, plural, identifier, platform) ->

    if typeof identifier is "number"
      singularIdKeyValue = {}
      singularIdKeyValue["#{singular}Id"] = identifier
      doc = Apollos[plural].findOne singularIdKeyValue

    else if typeof identifier is "string"
      doc = Apollos[plural].findOne identifier

    else
      doc = identifier

    if typeof doc isnt "object"
      throw new Meteor.Error "Delete Error", "Could not delete #{singular}
        identified by #{identifier}"

    if platform
      doc.updatedBy = platform.toUpperCase()
    else
      doc.updatedBy = Apollos.name

    # We have to update this first so the collection hooks know what to do
    Apollos[plural].update
      _id: doc._id
    ,
      $set:
        "updatedBy": doc.updatedBy

    Apollos[plural].remove doc._id
