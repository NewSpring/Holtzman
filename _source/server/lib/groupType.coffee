Apollos.groupType = {}


###

  Apollos.groupType.translate

  @example take data from a service and format it for Apollos

    Apollos.groupType.translate([obj, platform])

  @param groupType [Object] existing object to be translated
  @param platform [String] platform to be translated to

###
Apollos.groupType.translate = (groupType, platform) ->
  return Apollos.documentHelpers.translate "groupType", groupType, platform


###

  Apollos.groupType.update

  @example update a groupType in apollos with data from another service

    Apollos.groupType.update([obj, platform])

  @param groupType [Object] existing groupType from other service to be updated
  @param platform [String] platform to be update from

###
Apollos.groupType.update = (groupType, platform) ->
  return Apollos.documentHelpers.update "groupType", "groupTypes", groupType, platform


###

  Apollos.groupType.delete

  @example take a groupType and delete it

    Apollos.groupType.delete(user, [platform])

  @param groupType [Object|String|Number] existing document, _id
  @param platform [String] platform initiating the delete

###
Apollos.groupType.delete = (groupType, platform) ->
  Apollos.documentHelpers.delete "groupType", "groupTypes", groupType, platform
