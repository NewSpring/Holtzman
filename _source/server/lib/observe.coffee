changeWasAlreadyHandled = (doc, collection) ->

  if not doc.observationHash
    return false

  hash = hashObject doc

  if doc.observationHash is hash
    return true

  Apollos[collection].update doc._id, $set: observationHash: hash
  doc.observationHash = hash
  return false

hashObject = (obj) ->

  json = JSON.stringify obj
  hash = 0
  i = 0
  len = json.length

  if len is 0
    return hash

  while i < len
    chr = json.charCodeAt(i)
    hash  = ((hash << 5) - hash) + chr
    hash |= 0
    i++

  return hash

Apollos.observe = (collection) ->

  initializing = true

  Apollos[collection].find({},
    fields:
      observationHash: 0
      createdDate: 0
      updatedDate: 0
      updatedBy: 0
  ).observeChanges

    added: (id) ->

      if initializing
        return

      doc = Apollos[collection].findOne id
      if changeWasAlreadyHandled doc, collection
        Apollos.debug "#{collection} #{id} add already handled"
        return

      Apollos.debug "Handling add in #{collection} #{id}"
      Apollos[collection].added or= {}
      platformAddedBy = doc.updatedBy?.toUpperCase()

      for platform, handle of Apollos[collection].added
        if platformAddedBy and (platformAddedBy isnt platform.toUpperCase())
          handle doc

    changed: (id) ->

      doc = Apollos[collection].findOne id
      if changeWasAlreadyHandled doc, collection
        Apollos.debug "#{collection} #{id} change already handled"
        return

      Apollos.debug "Handling change in #{collection} #{id}"
      Apollos[collection].changed or= {}
      platformChangedBy = doc.updatedBy?.toUpperCase()

      for platform, handle of Apollos[collection].changed
        if platformChangedBy and (platformChangedBy isnt platform.toUpperCase())
          handle doc

    removed: (id) ->

      Apollos[collection].deleted or= {}

      for platform, handle of Apollos[collection].deleted
        handle id


  initializing = false



Apollos.observe.remove = (doc, platform, methods) ->

  if not Apollos[doc]
    Apollos.debug "Cannot remove observe methods for #{doc} becuase it doesn't exist"
    return

  if not platform
    Apollos.debug "Must specify platform"
    return

  if not methods
    methods = ["added", "changed", "removed"]

  for method in methods
    delete Apollos[doc][method]


Apollos.observe.add = (doc, platform, methods) ->

  if not Apollos[doc]
    Apollos.debug "Cannot set observe methods for #{doc} becuase it doesn't exist"
    return

  if not platform
    Apollos.debug "Must specify platform"
    return


  for method, handle of methods

    Apollos[doc][method] or= {}

    if Apollos[doc][method][platform]
      Apollos.debug "The #{method} observe handler for #{doc} from #{platform} has already been set"
      return
    Apollos[doc][method][platform] = handle


  return
