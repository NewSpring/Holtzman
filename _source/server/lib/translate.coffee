

Apollos.translate = (collection, platform, handler) ->

  if not Apollos[collection]
    Apollos.debug "Cannot set translate method for #{collection} becuase it doesn't exist"
    return

  if not platform
    Apollos.debug "Must specify platform"
    return

  platform = platform.toUpperCase()
  Apollos[collection]._dictionary or= {}

  if Apollos[collection]._dictionary[platform]
    Apollos.debug "The translation for #{collection} from #{platform} has already been set"
    return

  Apollos[collection]._dictionary[platform] = handler

  return
