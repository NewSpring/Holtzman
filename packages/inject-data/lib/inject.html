<script type="text/inject-data"><%= data %></script>
