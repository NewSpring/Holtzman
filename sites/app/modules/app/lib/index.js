

// named collections in EE :(
import Collections from "./collections"

export default {
  Collections
}
