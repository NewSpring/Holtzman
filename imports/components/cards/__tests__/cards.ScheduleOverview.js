import renderer from "react-test-renderer";
import { reset } from "aphrodite/lib/inject";
import ScheduleOverview, {
  getCurrencySymbol,
  getNegative,
  getDollars,
  getCents
} from "../cards.ScheduleOverview";

  jest.mock("moment", () => (date) => ({
  format: (style) => `${date || "now"}.format(${style})`,
}));

describe("ScheduleOverview", () => {
  beforeEach(() => {
    reset();
  });

  afterEach(() => {
    reset();
  });

  it("has an amount, fund, frequency, started, latest and onEditClick.", () => {
    const tree = renderer.create(
      <ScheduleOverview
        amount="$420.00"
        baseHeadingSize="2"
        fund="Step Up Fund"
        frequency="Once A Month"
        started="Tue Mar 15 2016 20:00:00 GMT-0400 (EDT)"
        latest="Tue Mar 16 2016 20:00:00 GMT-0400 (EDT)"
        onEditClick={jest.fn()}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it("correctly displays a negative value.", () => {
    const tree = renderer.create(
      <ScheduleOverview
        amount="$-420.00"
        baseHeadingSize="2"
        fund="Step Up Fund"
        frequency="Once A Month"
        started="Tue Mar 15 2016 20:00:00 GMT-0400 (EDT)"
        latest="Tue Mar 16 2016 20:00:00 GMT-0400 (EDT)"
        onEditClick={jest.fn()}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it("doesn't have a latest transaction, if one isn't passed", () => {
    const tree = renderer.create(
      <ScheduleOverview
        amount="$420.00"
        baseHeadingSize="2"
        fund="Step Up Fund"
        frequency="Once A Month"
        started="Tue Mar 15 2016 20:00:00 GMT-0400 (EDT)"
        onEditClick={jest.fn()}
      />
    )

    expect(tree).toMatchSnapshot();
  });

  it("falls back correctly if currency symbol isn't present.", () => {
    const tree = renderer.create(
      <ScheduleOverview
        amount="420.00"
        baseHeadingSize="2"
        fund="Step Up Fund"
        frequency="Once A Month"
        started="Tue Mar 15 2016 20:00:00 GMT-0400 (EDT)"
        onEditClick={jest.fn()}
      />
    )

    expect(tree).toMatchSnapshot();
  });

  it("falls back correctly if there isn't any cents.", () => {
    const tree = renderer.create(
      <ScheduleOverview
        amount="$420"
        baseHeadingSize="2"
        fund="Step Up Fund"
        frequency="Once A Month"
        started="Tue Mar 15 2016 20:00:00 GMT-0400 (EDT)"
        onEditClick={jest.fn()}
      />
    )

    expect(tree).toMatchSnapshot();
  });

  it("falls back correctly if there isn't an amount.", () => {
    const tree = renderer.create(
      <ScheduleOverview
        amount=""
        baseHeadingSize="2"
        fund="Step Up Fund"
        frequency="Once A Month"
        started="Jun 15, 2015"
        onEditClick={jest.fn()}
      />
    )

    expect(tree).toMatchSnapshot();
  });

  it("falls back correctly if a base heading size isn't set", () => {
    const tree = renderer.create(
      <ScheduleOverview
        amount="$420.00"
        fund="Step Up Fund"
        frequency="Once A Month"
        started="Tue Mar 15 2016 20:00:00 GMT-0400 (EDT)"
        onEditClick={jest.fn()}
      />
    )

    expect(tree).toMatchSnapshot();
  });
});
