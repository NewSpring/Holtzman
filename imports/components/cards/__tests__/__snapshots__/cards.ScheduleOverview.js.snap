exports[`ScheduleOverview correctly displays a negative value. 1`] = `
<div
  className="card">
  <div
    className="card__item soft push-half-ends one-whole">
    <button
      className="float-right h6 text-dark-tertiary outlined--bottom"
      onClick={[Function]}
      style={
        Object {
          "borderColor": "inherit",
          "borderWidth": "2px",
        }
      }>
      Edit
    </button>
    <div
      className="floating text-left text-dark-primary">
      <h4
        className="floating__item flush"
        style={
          Object {
            "paddingRight": "5px",
          }
        }>
        $
      </h4>
      <h4
        className="floating__item flush"
        style={
          Object {
            "paddingRight": "3px",
          }
        }>
        -
      </h4>
      <h2
        className="floating__item flush">
        420
      </h2>
      <h4
        className="floating__item flush">
        .
        00
      </h4>
    </div>
    <div
      className="floating text-left push-bottom">
      <h5
        className="floating__item soft-half-right flush text-dark-primary">
        Step Up Fund
      </h5>
      <h6
        className="floating__item flush text-dark-tertiary">
        Once A Month
      </h6>
    </div>
    <p
      className="flush"
      style={
        Object {
          "marginBottom": "5px",
        }
      }>
      <span
        className="h7 text-dark-tertiary push-half-right"
        style={
          Object {
            "verticalAlign": "middle",
          }
        }>
        Started: 
      </span>
      <em
        className="text-dark-primary">
        Jun 15, 2015
      </em>
    </p>
    <p
      className="flush">
      <span
        className="h7 text-dark-tertiary push-half-right"
        style={
          Object {
            "verticalAlign": "middle",
          }
        }>
        Latest: 
      </span>
      <em
        className="text-dark-primary">
        Nov 15, 2016
      </em>
    </p>
  </div>
</div>
`;

exports[`ScheduleOverview doesn't have a latest transaction, if one isn't passed 1`] = `
<div
  className="card">
  <div
    className="card__item soft push-half-ends one-whole">
    <button
      className="float-right h6 text-dark-tertiary outlined--bottom"
      onClick={[Function]}
      style={
        Object {
          "borderColor": "inherit",
          "borderWidth": "2px",
        }
      }>
      Edit
    </button>
    <div
      className="floating text-left text-dark-primary">
      <h4
        className="floating__item flush"
        style={
          Object {
            "paddingRight": "5px",
          }
        }>
        $
      </h4>
      <h2
        className="floating__item flush">
        420
      </h2>
      <h4
        className="floating__item flush">
        .
        00
      </h4>
    </div>
    <div
      className="floating text-left push-bottom">
      <h5
        className="floating__item soft-half-right flush text-dark-primary">
        Step Up Fund
      </h5>
      <h6
        className="floating__item flush text-dark-tertiary">
        Once A Month
      </h6>
    </div>
    <p
      className="flush"
      style={Object {}}>
      <span
        className="h7 text-dark-tertiary push-half-right"
        style={
          Object {
            "verticalAlign": "middle",
          }
        }>
        Started: 
      </span>
      <em
        className="text-dark-primary">
        Jun 15, 2015
      </em>
    </p>
  </div>
</div>
`;

exports[`ScheduleOverview falls back correctly if currency symbol isn't present. 1`] = `
<div
  className="card">
  <div
    className="card__item soft push-half-ends one-whole">
    <button
      className="float-right h6 text-dark-tertiary outlined--bottom"
      onClick={[Function]}
      style={
        Object {
          "borderColor": "inherit",
          "borderWidth": "2px",
        }
      }>
      Edit
    </button>
    <div
      className="floating text-left text-dark-primary">
      <h4
        className="floating__item flush"
        style={
          Object {
            "paddingRight": "5px",
          }
        }>
        $
      </h4>
      <h2
        className="floating__item flush">
        420
      </h2>
      <h4
        className="floating__item flush">
        .
        00
      </h4>
    </div>
    <div
      className="floating text-left push-bottom">
      <h5
        className="floating__item soft-half-right flush text-dark-primary">
        Step Up Fund
      </h5>
      <h6
        className="floating__item flush text-dark-tertiary">
        Once A Month
      </h6>
    </div>
    <p
      className="flush"
      style={Object {}}>
      <span
        className="h7 text-dark-tertiary push-half-right"
        style={
          Object {
            "verticalAlign": "middle",
          }
        }>
        Started: 
      </span>
      <em
        className="text-dark-primary">
        Jun 15, 2015
      </em>
    </p>
  </div>
</div>
`;

exports[`ScheduleOverview falls back correctly if there isn't an amount. 1`] = `
<div
  className="card">
  <div
    className="card__item soft push-half-ends one-whole">
    <button
      className="float-right h6 text-dark-tertiary outlined--bottom"
      onClick={[Function]}
      style={
        Object {
          "borderColor": "inherit",
          "borderWidth": "2px",
        }
      }>
      Edit
    </button>
    <div
      className="floating text-left text-dark-primary">
      <h4
        className="floating__item flush"
        style={
          Object {
            "paddingRight": "5px",
          }
        }>
        $
      </h4>
      <h2
        className="floating__item flush">
        0
      </h2>
      <h4
        className="floating__item flush">
        .
        00
      </h4>
    </div>
    <div
      className="floating text-left push-bottom">
      <h5
        className="floating__item soft-half-right flush text-dark-primary">
        Step Up Fund
      </h5>
      <h6
        className="floating__item flush text-dark-tertiary">
        Once A Month
      </h6>
    </div>
    <p
      className="flush"
      style={Object {}}>
      <span
        className="h7 text-dark-tertiary push-half-right"
        style={
          Object {
            "verticalAlign": "middle",
          }
        }>
        Started: 
      </span>
      <em
        className="text-dark-primary">
        Jun 15, 2015
      </em>
    </p>
  </div>
</div>
`;

exports[`ScheduleOverview falls back correctly if there isn't any cents. 1`] = `
<div
  className="card">
  <div
    className="card__item soft push-half-ends one-whole">
    <button
      className="float-right h6 text-dark-tertiary outlined--bottom"
      onClick={[Function]}
      style={
        Object {
          "borderColor": "inherit",
          "borderWidth": "2px",
        }
      }>
      Edit
    </button>
    <div
      className="floating text-left text-dark-primary">
      <h4
        className="floating__item flush"
        style={
          Object {
            "paddingRight": "5px",
          }
        }>
        $
      </h4>
      <h2
        className="floating__item flush">
        420
      </h2>
      <h4
        className="floating__item flush">
        .
        00
      </h4>
    </div>
    <div
      className="floating text-left push-bottom">
      <h5
        className="floating__item soft-half-right flush text-dark-primary">
        Step Up Fund
      </h5>
      <h6
        className="floating__item flush text-dark-tertiary">
        Once A Month
      </h6>
    </div>
    <p
      className="flush"
      style={Object {}}>
      <span
        className="h7 text-dark-tertiary push-half-right"
        style={
          Object {
            "verticalAlign": "middle",
          }
        }>
        Started: 
      </span>
      <em
        className="text-dark-primary">
        Jun 15, 2015
      </em>
    </p>
  </div>
</div>
`;

exports[`ScheduleOverview has an amount, fund, frequency, started, latest and onEditClick. 1`] = `
<div
  className="card">
  <div
    className="card__item soft push-half-ends one-whole">
    <button
      className="float-right h6 text-dark-tertiary outlined--bottom"
      onClick={[Function]}
      style={
        Object {
          "borderColor": "inherit",
          "borderWidth": "2px",
        }
      }>
      Edit
    </button>
    <div
      className="floating text-left text-dark-primary">
      <h4
        className="floating__item flush"
        style={
          Object {
            "paddingRight": "5px",
          }
        }>
        $
      </h4>
      <h2
        className="floating__item flush">
        420
      </h2>
      <h4
        className="floating__item flush">
        .
        00
      </h4>
    </div>
    <div
      className="floating text-left push-bottom">
      <h5
        className="floating__item soft-half-right flush text-dark-primary">
        Step Up Fund
      </h5>
      <h6
        className="floating__item flush text-dark-tertiary">
        Once A Month
      </h6>
    </div>
    <p
      className="flush"
      style={
        Object {
          "marginBottom": "5px",
        }
      }>
      <span
        className="h7 text-dark-tertiary push-half-right"
        style={
          Object {
            "verticalAlign": "middle",
          }
        }>
        Started: 
      </span>
      <em
        className="text-dark-primary">
        Jun 15, 2015
      </em>
    </p>
    <p
      className="flush">
      <span
        className="h7 text-dark-tertiary push-half-right"
        style={
          Object {
            "verticalAlign": "middle",
          }
        }>
        Latest: 
      </span>
      <em
        className="text-dark-primary">
        Nov 15, 2016
      </em>
    </p>
  </div>
</div>
`;
